### ISSUE TITLE
[BUG] Export tiddler to static html creates extra p tag (paragraph)

### ISSUE THREAD

**Issue by kookma** [Jermolene/TiddlyWiki5#5617]
**Describe the bug**
While working on Refnotes plugin I noticed that, export a tiddler to static html file generate extra _p_ tag! 

**To Reproduce**
1. Go to https://kookma.github.io/TW-Refnotes/#demo%2Fref-macro
2. See the Example. You see this text `Here a citation is made to Fátima Leal, et al. (2018) . Another reference is made to Stefan Bosse (2018) and the third reference is made to N.E. Amadeo and M.A. Laborde (1995) .` One paragraph with two lines!
3. You can use the browser inspect and see `Here a citation is made to`  like `"Here a citation is made to" ` 
4. Now export this tiddler (https://kookma.github.io/TW-Refnotes/#demo%2Fref-macro) as static html file
5. Look at the Example! the `Here a citation is made to` is rendered as `<p>`Here a citation is made to</p>` You can use browser inspect to see this new generated p tag. This was not present in TW.

**Expected behavior**
What we see in TW in rendered mode shall be the same in exported static html file.

**Screenshots**

**Fig 1. In Tiddlywiki (before export to html)**
![image](https://user-images.githubusercontent.com/830394/115259548-3cd0d100-a147-11eb-8db8-414d7db5a98c.png)

**Fig 2. Exported tiddler (after export to html)**
![img_81_chrome](https://user-images.githubusercontent.com/830394/115259663-52de9180-a147-11eb-9891-3dcef1d9dae8.png)

**TiddlyWiki Configuration (please complete the following information):**
 - Version 5.1.24
 - Saving mechanism online
 - Plugins installed Refnotes
 
**Desktop (please complete the following information):**
 - OS: Windows 10
 - Browser Chrome 89

**Comment by kookma** [comment]
Further information:

I installed the official internals and tried to replicate the issue. This is the part of RAW HTML returned by Internals plugin!

```
<p>Here a citation is made to 
<div class="refnotes-tooltip refcls">
<span class=" tc-reveal" hidden="true"></span>
<span class=" tc-reveal">
Fátima Leal, et al. (2018)
<div class="refnotes-tooltiptext">
```

This is wat I get from Browser Inspect on Exported Static Html tiddler!

![image](https://user-images.githubusercontent.com/830394/115263595-cb931d00-a14a-11eb-879e-d5bbab9f6bf5.png)


I am not sure what happen here!

**Comment by kookma** [comment]
> The issue is that an opening HTML tag followed by two blank lines will always trigger block mode for the parsing of the content of that element.

That's correct!

So, you recommend to revise code to prevent such situation! May be using \whitespace trim and deleting extra lines in code! Am I right?

**Comment by ght** [comment]
I strongly suspect that this is just an unexpected detail of the HTML standard catching you offguard.

## TL;DR
The `</p>` will be inserted automatically in certain cases:
> A p element's end tag can be omitted if the p element is immediately followed by an (…) div, (…) element, or if there is no more content in the parent element (…)
-- [HTML Standard](https://html.spec.whatwg.org/multipage/grouping-content.html#the-p-element)

The following minimal example demonstrates how the browser automatically closes the p element: Navigate to `data:text/html,<p>a<div>b</div>c</p>` (just paste it into the address bar and hit enter) and inspect the DOM in your dev tools; it will look like `<p>a</p><div>b</div>c<p></p>`.

## Some more details:
The [p element](https://html.spec.whatwg.org/multipage/grouping-content.html#the-p-element)'s content model is [*phrasing content*](https://html.spec.whatwg.org/multipage/dom.html#phrasing-content), simply put "inline elements".

The [div element](https://html.spec.whatwg.org/multipage/grouping-content.html#the-div-element) on the other hand is [*flow content*](https://html.spec.whatwg.org/multipage/dom.html#flow-content) (more or less "blocks"), so it cannot appear as the child of an element which only allows *phrasing content*.

In simpler terms: `p` and `div` are blocks, and while `div` explicitly allows further blocks (*flow content*) to appear as children, the `p` does not and only allows inline elements (*phrasing content*). So when you open another block, the browser helpfully (or in this case confusingly) closes the `p`, since it knows that the standard does not allow blocks inside `p`.

**Comment by pmario** [comment]
IMO related to:  WIP Test p tag fix v5.4.0 #9146
